### PR TITLE
docs: Add storage change status analysis

### PR DESCRIPTION
**Pull request by chfast**
Naming these is fun...

There are also differences in basic names in EVMC and the EVM spec:
- `ADDED` vs `SET`,
- `MODIFIED` vs `RESET`,
- `DELETED` vs `CLEARS`.

### CODE REVIEW

File: docs/storage_change_status.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+## Storage change statuses
+
+- `0` - a zero storage value,
+- `X` - non-zero storage value,
+- `Y` - non-zero storage value different from `X`,
+- `Z` - non-zero storage value different form `X` and `Y`,
+- `...` - any value.
+
+
+<table>
```
Comment (axic) [inline docs/storage_change_status.md]: Why not use a markdown table?

This one can format it for you: https://www.tablesgenerator.com/markdown_tables

Comment (chfast) [reply docs/storage_change_status.md]: No rowspans.

File: docs/storage_change_status.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
+# Storage change status
+
+> Analysis of possible storage change statuses in [EIP-2200] / [EIP-1283].
+
+## Cost constants
+
+| Constant | Value |
+| -------- | ----- |
+| SLOAD    | 200   |
```
Comment (holiman) [inline docs/storage_change_status.md]: You're using `200` for all of these - but it's `800` no?

Comment (chfast) [reply docs/storage_change_status.md]: Yes, the **unmerged** EIP-2200 uses 800, but I wasn't sure the repricing 100% confirmed to go though. The precise value is secondary in the context of this document.

File: docs/storage_change_status.md (hunk 1 above)
Comment (axic) [reply docs/storage_change_status.md]: Never mind, found the way to "display file" so that it renders.

File: docs/storage_change_status.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            <td>yes</td>
+            <td>200 = SLOAD</td>
+            <td>4800 = RESET - SLOAD</td>
+        </tr>
+    </tbody>
+</table>
+
+
+[EIP-1283]: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1283.md
+[EIP-2200]: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2200.md
```
Comment (axic) [inline docs/storage_change_status.md]: Perhaps use https://eips.ethereum.org/EIPS/eip-1283, etc.?

**Comment by axic** [comment]
@chfast can this be merged (without any major changes) or this current version is invalid?

File: docs/storage_change_status.md (hunk 2 above)
Comment (axic) [reply docs/storage_change_status.md:9]: It did go through, eventually, haha :)

**Comment by axic** [comment]
Rebased. And trying updating it with the final Istanbul numbers.

**Comment by axic** [comment]
This has doxygen warnings:
```
/home/builder/project/docs/storage_change_status.md:29: warning: expected <tr> or <caption> tag but found <thead> instead!
/home/builder/project/docs/storage_change_status.md:39: warning: Unsupported xml/html tag </thead> found
/home/builder/project/docs/storage_change_status.md:40: warning: Unsupported xml/html tag <tbody> found
/home/builder/project/docs/storage_change_status.md:137: warning: Unsupported xml/html tag </tbody> found
```

**Pull request closed without merging**